### ISSUE TITLE
Failed to create models in the cloud: Modelgen job creation failed

### ISSUE THREAD

**Issue by gspeicher** [aws-amplify/amplify-console#1640]
**Describe the bug**
There seems to be something going on with our Amplify backend and it is for all practical purposes impossible to understand what is wrong (or why) using the available tools.

`amplify push` on our existing project is ending its output with:
```
Generating models in the cloud…
Failed to create models in the cloud: Modelgen job creation failed
All resources are updated in the cloud
```

If I open the Amplify Admin UI and look at Deployment Activity, it shows that `jobCategory.GenerateBackendAPIModels` failed but does not give any reason for the failure. Within the Admin UI if I click on "View data model" the entire web app just goes full white screen so this is no help either.

**To Reproduce**
Hard to say exactly how to reproduce this from a brand new project, but for our existing project I can use AWS CLI to run `aws amplifybackend generate-backend-api-models` and check the job status to see that it failed with no error message. Or I can continue to use `amplify push` to yield the same results. I have also tried using `--force` to no avail.

For what it's worth, the push was one of several related to the modification of @key annotations on secondary indices since you have to do them in separate steps.

**Expected behavior**
At a minimum, there should be an error message sufficient to describe what is wrong with our backend and provide some clue how to fix it. Ideally it would not be possible to get our backend into a broken state like this.

**Additional context**
The app ID is "d2r3gwm433d7n4" and this is environment "master"

This is happening with Amplify CLI v4.45.0 and 4.45.1 as well.

**Comment by gspeicher** [comment]
Edited to remove paragraph about problems logging in to Amplify Admin UI, which was caused a link copy/paste issue. But have confirmed that attempting to view the data model with the Admin UI still fails in a second browser.

**Comment by lewistham9x** [comment]
I am having the same issue as well with my app ID "d2zy4wzrz947nj" on "staging"

**Comment by wezside** [comment]
Same issue here
App ID is ds7cntrpahvdv and environment is "dev"

**Comment by inspiran** [comment]
yep, same issue here.

**Comment by swaminator** [comment]
@inspiran @wezside @lewistham9x @gspeicher Can you guys try again?

**Comment by inspiran** [comment]
mine is working again @swaminator .  Curious what you did.  Thx for fixing it so quickly.

**Comment by kaustavghosh06** [comment]
@gspeicher Which region are you in?

**Comment by gspeicher** [comment]
us-east-1

**Comment by kaustavghosh06** [comment]
@gspeicher I looked at the service logs and found out the following:
`missing the following inputParams signinwithappleAppIdUserPool, signinwithappleAppSecretUserPool`
The Amplify CLI doesn't support Sign-in with apple yet, what are the changes that you've made to enable this in your Amplify Project? This might help us debug further.

**Comment by kaustavghosh06** [comment]
@inspiran We fixed a bug in our service yesterday afternoon which might've resulted in the flow working for you now.

**Comment by gspeicher** [comment]
We made Apple sign-in work via modifications to the CloudFormation template files for our auth resource, and to the value for "hostedUIProviderCreds" in our deployment secrets file. Kind of surprising that generating models would be related to the presence of Apple sign-in secrets but those two params definitely do not exist.

Incidentally, if you have a better way to support Apple sign-in then I am all ears. Our CI/CD pipeline is also broken because there is no support for this env variable in the Amplify console.

**Comment by kaustavghosh06** [comment]
@gspeicher Yeah, the generation of models is for the Admin CMS to work correctly and that happens in our service (in the cloud). Your local models would still be created. In our service we run `amplify pull` to pull down your project configurations and then run codegen to generate data models. It's for the same reason that your CI/CD might be broken since there isn't any env variable/headless support for Apple Sign-in yet.

**Comment by gspeicher** [comment]
Can you confirm whether this would have any operational impact on our system or if it is strictly limited to an “informational error” on deployment and a broken Admin UI?

**Comment by kaustavghosh06** [comment]
Yeah, this would only impact Amplify Console CI/CD and Admin UI. The CLI would still functionally work and has no impact with this.

**Comment by gspeicher** [comment]
Thanks. I guess we can close this and maybe reference it to the Apple sign-in support issue.

**Comment by emmafass** [comment]
@kaustavghosh06 I came across this thread and would love your input in my problem. I don't use any Apple sign in, but I still am getting the same error as @gspeicher. 

Mine started happening after the CLI prompted me to upgrade my lambda functions (which I set up via the CLI) from node v10 to node v12. 

I describe the reproduction steps [here](https://github.com/aws-amplify/amplify-cli/issues/6959 )

(Let me know if I should close this ticket on the amplify-cli repo. I thought that would be the correct repo but maybe console is correct.)
